### PR TITLE
replace fmt.sprintf

### PR DESCRIPTION
**Pull request by vdepatla**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12728

### CODE REVIEW

File: app/ldap.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func (a *App) GetAllLdapGroupsPage(page int, perPage int, opts model.LdapGroupSe
 		}
 	} else {
 		ae := model.NewAppError("GetAllLdapGroupsPage", "ent.ldap.app_error", nil, "", http.StatusNotImplemented)
-		mlog.Error(fmt.Sprintf("%v", ae.Error()))
+		mlog.Error("LDAP groups not found for the configired group", mlog.Any(opts))
```
Comment (jespino) [inline app/ldap.go]: This is not the error expected here, actually the error should be something like `mlog.Error("Unable to use ldap", mlog.Err(err))`

File: app/ldap.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func (a *App) GetLdapGroup(ldapGroupID string) (*model.Group, *model.AppError) {
 		}
 	} else {
 		ae := model.NewAppError("GetLdapGroup", "ent.ldap.app_error", nil, "", http.StatusNotImplemented)
-		mlog.Error(fmt.Sprintf("%v", ae.Error()))
+		mlog.Error("LDAP group not found", mlog.String(ldapGroupID))
```
Comment (jespino) [inline app/ldap.go]: Same here, it should be something like `mlog.Error("Unable to use ldap", mlog.Err(err))`

Comment (vdepatla) [reply app/ldap.go]: UPDATED

File: app/ldap.go (hunk 1 above)
Comment (vdepatla) [reply app/ldap.go]: UPDATED

File: app/ldap.go (hunk 2 above)
Comment (marianunez) [reply app/ldap.go]: @vaithak The error in this case is that Ldap is not available to even do the get operation so I believe a better message would be:
 
```suggestion
		mlog.Error("Unable to use ldap", mlog.Error(ae))
```

File: app/ldap.go (hunk 1 above)
Comment (marianunez) [reply app/ldap.go]: Same here: 

> 
```suggestion
		mlog.Error("Unable to use ldap", mlog.Error(ae))
```

File: app/ldap.go (hunk 2 above)
Comment (vdepatla) [reply app/ldap.go]: @marianunez what is the suggested change ? is there an updated message ?

File: app/ldap.go [hunk 3]
```diff
@@ -52,7 +52,7 @@ func (a *App) GetLdapGroup(ldapGroupID string) (*model.Group, *model.AppError) {
 		}
 	} else {
 		ae := model.NewAppError("GetLdapGroup", "ent.ldap.app_error", nil, "", http.StatusNotImplemented)
-		mlog.Error(fmt.Sprintf("%v", ae.Error()))
+		mlog.Error("Unable to use ldap", mlog.String(ldapGroupID))
```
Comment (lieut-data) [inline app/ldap.go]: @vdepatla, the behaviour change here is unexpected: we're no longer logging the error at all. Can we include it?
```suggestion
		mlog.Error("Unable to use ldap", mlog.String("ldap_group_id", ldapGroupID), mlog.Err(ae))
```

File: app/ldap.go [hunk 4]
```diff
@@ -73,7 +73,7 @@ func (a *App) GetAllLdapGroupsPage(page int, perPage int, opts model.LdapGroupSe
 		}
 	} else {
 		ae := model.NewAppError("GetAllLdapGroupsPage", "ent.ldap.app_error", nil, "", http.StatusNotImplemented)
-		mlog.Error(fmt.Sprintf("%v", ae.Error()))
+		mlog.Error("Unable to use ldap", mlog.Any(opts))
```
Comment (lieut-data) [inline app/ldap.go]: I'm not sure it makes sense to log `opts` here, at least not without `page` and `perPage`. I'd suggest just logging the original error:
```suggestion
		mlog.Error("Unable to use ldap", mlog.Err(ae))
```

**Pull request merged**